modules/tensor_mechanics/src/materials/ComputeAdmissibleState.C

```diff
@@ -0,0 +1,281 @@
+                                     "inelastic_models.  Default = '1 1 ... 1'.  This "
+                                     "parameter is set to 1 if the number of models = 1");
+  return params;
+}
+
+ComputeAdmissibleState::ComputeAdmissibleState(const InputParameters & parameters)
+  : ComputeFiniteStrainElasticStress(parameters),
+    _max_its(parameters.get<unsigned int>("max_iterations")),
+    _relative_tolerance(parameters.get<Real>("relative_tolerance")),
+    _absolute_tolerance(parameters.get<Real>("absolute_tolerance")),
+    _output_iteration_info(getParam<bool>("output_iteration_info")),
+    _perform_finite_strain_rotations(getParam<bool>("perform_finite_strain_rotations")),
+    _elasticity_tensor(getMaterialPropertyByName<RankFourTensor>(_base_name + "elasticity_tensor")),
+    _elastic_strain_old(getMaterialPropertyOld<RankTwoTensor>(_base_name + "elastic_strain")),
+    _strain_increment(getMaterialProperty<RankTwoTensor>(_base_name + "strain_increment")),
+    _inelastic_strain(declareProperty<RankTwoTensor>(_base_name + "combined_inelastic_strain")),
+    _inelastic_strain_old(
+        getMaterialPropertyOld<RankTwoTensor>(_base_name + "combined_inelastic_strain")),
+    _tangent_operator_type(getParam<MooseEnum>("tangent_operator").getEnum<TangentOperatorEnum>()),
+    _num_models(getParam<std::vector<MaterialName>>("inelastic_models").size()),
+    _inelastic_weights(isParamValid("combined_inelastic_strain_weights")
+                           ? getParam<std::vector<Real>>("combined_inelastic_strain_weights")
+                           : std::vector<Real>(_num_models, true)),
+    _cto(_num_models)
+{
+  if (_inelastic_weights.size() != _num_models)
+    mooseError("ComputeAdmissibleState: combined_inelastic_strain_weights must contain the same "
+               "number of entries as inelastic_models ",
+               _inelastic_weights.size(),
+               " ",
+               _num_models);
+}
+
+void
+ComputeAdmissibleState::initQpStatefulProperties()
+{
+  ComputeStressBase::initQpStatefulProperties();
+  _inelastic_strain[_qp].zero();
+}
+
+void
+ComputeAdmissibleState::initialSetup()
+{
+  std::vector<MaterialName> models = getParam<std::vector<MaterialName>>("inelastic_models");
+  for (unsigned int i = 0; i < _num_models; ++i)
+  {
+    StateUpdateBase * rrr = dynamic_cast<StateUpdateBase *>(&getMaterialByName(models[i]));
+    if (rrr)
+      _models.push_back(rrr);
+    else
+      mooseError("Model " + models[i] + " is not compatible with ComputeAdmissibleState");
+  }
+}
+
+void
+ComputeAdmissibleState::computeQpStress()
+{
+  // Nothing to update during the first time step
+  // ComputeQpStress is not called during the zeroth time step, so no need to guard against _t_step
+  // == 0
+
+  RankTwoTensor elastic_strain_increment;
+  RankTwoTensor combined_inelastic_strain_increment;
+
+  if (_num_models == 1)
+    updateQpStateSingleModel(elastic_strain_increment, combined_inelastic_strain_increment);
```

Comment:
@WilkAndy I've got a couple of Bison materials which inherit from RadialReturnStressUpdate but override the StateUpdate method.  When I'm using more than one of these overriding-StateUpdate materials, everything is fine and dandy, but when I use just a single model and use the `updateQpStateSingleModel` I hit a runtime error.  With the debugger the error occurs in libmesh:

```
frame #0: 0x00000001001c26df libbison-dbg.0.dylib`libMesh::TypeVector<libMesh::CompareTypes<double, double>::supertype> libMesh::TypeVector<double>::operator-<double>(this=0x000000010ad9e280, p=0x000000010ad9e7a0) const at type_vector.h:643
   640 	#endif
   641 	
   642 	#if LIBMESH_DIM == 3
-> 643 	  return TypeVector<TS>(_coords[0] - p._coords[0],
   644 	                        _coords[1] - p._coords[1],
   645 	                        _coords[2] - p._coords[2]);
   646 	#endif
```
Haven't hit this error with any of the other 'normal' RadialReturnStressUpdate materials, so it must be something with my odd Bison materials.  I'll keep working on this tomorrow, but if you ran into a similar problem with the other radial return models and could point me in the right direction, I'd greatly appreciate the help.  Thanks!